cli/src/semgrep/formatter/sarif.py

```diff
@@ -130,8 +181,9 @@ def _dataflow_trace_to_thread_flows_sarif(rule_match: RuleMatch) -> Any:
         intermediate_vars = dataflow_trace.intermediate_vars
 
         if taint_source:
-            locations.append(
-                SarifFormatter._taint_source_to_thread_flow_location_sarif(rule_match)
+            # calculate intermediate vars/calls for the source
+            locations += SarifFormatter._taint_source_to_thread_flow_location_sarif(
```

Comment:
Previously, `SarifFormatter._taint_source_to_thread_flow_location_sarif` was returning the thread-flow location of the source.
Now it's returning a list of thread-flow locations explaining how the source is tainted through a call-chain.
Please update the name.
```suggestion
            locations += SarifFormatter._taint_source_to_thread_flow_locations_sarif(
```

Reply:
Done